src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -20,38 +20,60 @@
 SET_TTL = 24 * 60 * 60
 
 
+REDIS_KEY_PREFIX = "txnames:"
+
 add_to_set = redis.load_script("utils/sadd_capped.lua")
 
 
 def _get_redis_key(project: Project) -> str:
-    return f"txnames:o:{project.organization_id}:p:{project.id}"
+    return f"{REDIS_KEY_PREFIX}o:{project.organization_id}:p:{project.id}"
 
 
-def _get_redis_client() -> Any:
+def get_redis_client() -> Any:
     cluster_key = getattr(settings, "SENTRY_TRANSACTION_NAMES_REDIS_CLUSTER", "default")
     return redis.redis_clusters.get(cluster_key)
 
 
+def _get_all_keys() -> Iterable[str]:
+    client = get_redis_client()
+    return client.scan_iter(match=f"{REDIS_KEY_PREFIX}*")  # type: ignore
```

Comment:
Is the `scan_iter` call instrumented? Can we detect bottlenecks here when running the task?